apoc.import.csv fails when dash is present in id space

## Guidelines

Please note that GitHub issues are only meant for bug reports/feature requests. If you have questions on how to use APOC, please ask on [the Neo4j Discussion Forum](https://community.neo4j.com/c/neo4j-graph-platform/procedures-apoc) instead of creating an issue here.

## Expected Behavior (Mandatory)

Given the following csv: (as per the header format here: https://neo4j.com/docs/operations-manual/current/tools/import/file-header-format/#import-tool-id-spaces)

```
personId:ID(Person-ID),name:STRING
1,"Alice"
2,"Bob"
```

The procedure will throw an error:

```
Failed to invoke procedure `apoc.import.csv`: Caused by: java.lang.IllegalStateException: No match found
```

## Actual Behavior (Mandatory)

By switching the headers in the csv to use an underscore instead of dash in the id space, the import succeeds:

```
personId:ID(Person_ID),name:STRING
```

## How to Reproduce the Problem

Call apoc.import.csv on the file given

### Simple Dataset (where it's possibile)

```
personId:ID(Person-ID),name:STRING
1,"Alice"
2,"Bob"

```


### Steps (Mandatory)

  1. Call apoc.import.csv on the given dataset through neo4j remote interface

## Screenshots (where it's possibile)

## Specifications (Mandatory)

Currently used versions

### Versions

  - OS: MacOS Catalina 10.15.2
  - Neo4j: 3.5.14 docker image
  - Neo4j-Apoc: 3.5

--- a/src/main/java/apoc/export/csv/CsvLoaderConstants.java
+++ b/src/main/java/apoc/export/csv/CsvLoaderConstants.java
@@ -4,7 +4,7 @@ import java.util.regex.Pattern;
 
 public class CsvLoaderConstants {
 
-    public static final Pattern FIELD_PATTERN = Pattern.compile("^(?<name>[^:]*)(:(?<type>\\w+))?(\\((?<idspace>\\w+)\\))?(?<array>\\[\\])?$");
+    public static final Pattern FIELD_PATTERN = Pattern.compile("^(?<name>[^:]*)(:(?<type>\\w+))?(\\((?<idspace>[-a-zA-Z_0-9]+)\\))?(?<array>\\[\\])?$");
     public static final String ARRAY_PATTERN = "[]";
 
     public static final String IGNORE_FIELD = "IGNORE";

--- a/src/test/java/apoc/export/csv/ImportCsvTest.java
+++ b/src/test/java/apoc/export/csv/ImportCsvTest.java
@@ -61,6 +61,9 @@ public class ImportCsvTest {
                     new AbstractMap.SimpleEntry<>("id-idspaces", ":ID(Person)|name:STRING\n" +
                             "1|John\n" +
                             "2|Jane\n"),
+                    new AbstractMap.SimpleEntry<>("id-idspaces-with-dash", ":ID(Person-Id)|name:STRING\n" +
+                            "1|John\n" +
+                            "2|Jane\n"),
                     new AbstractMap.SimpleEntry<>("id", "id:ID|name:STRING\n" +
                             "1|John\n" +
                             "2|Jane\n"),
@@ -155,6 +158,25 @@ public class ImportCsvTest {
                 "CALL apoc.import.csv([{fileName: $file, labels: ['Person']}], [], $config)",
                 map(
                         "file", "file:/id-idspaces.csv",
+                        "config", map("delimiter", '|')
+                ),
+                (r) -> {
+                    assertEquals(2L, r.get("nodes"));
+                    assertEquals(0L, r.get("relationships"));
+                }
+        );
+
+        List<String> names = TestUtil.firstColumn(db, "MATCH (n:Person) RETURN n.name AS name ORDER BY name");
+        assertThat(names, Matchers.containsInAnyOrder("Jane", "John"));
+    }
+
+    @Test
+    public void testNodesWithIdSpacesWithDash() {
+        TestUtil.testCall(
+                db,
+                "CALL apoc.import.csv([{fileName: $file, labels: ['Person']}], [], $config)",
+                map(
+                        "file", "file:/id-idspaces-with-dash.csv",
                         "config", map("delimiter", '|')
                 ),
                 (r) -> {

